src/ray/raylet/node_manager.cc

```diff
@@ -351,14 +365,45 @@ void NodeManager::HeartbeatAdded(gcs::AsyncGcsClient *client, const ClientID &cl
                   << client_id;
     return;
   }
-  SchedulingResources &resources = it->second;
-  ResourceSet heartbeat_resource_available(heartbeat_data.resources_available_label,
-                                           heartbeat_data.resources_available_capacity);
-  resources.SetAvailableResources(
+  SchedulingResources &remote_resources = it->second;
+
+  ResourceSet remote_available(heartbeat_data.resources_available_label,
+                               heartbeat_data.resources_available_capacity);
+  ResourceSet remote_load(heartbeat_data.resource_load_label,
+                                      heartbeat_data.resource_load_capacity);
+  // TODO(atumanov): assert that the load is a non-empty ResourceSet.
+  RAY_LOG(DEBUG) << "[HeartbeatAdded]: received load: "
+                << remote_load.ToString();
+  remote_resources.SetAvailableResources(
       ResourceSet(heartbeat_data.resources_available_label,
                   heartbeat_data.resources_available_capacity));
-  RAY_CHECK(this->cluster_resource_map_[client_id].GetAvailableResources() ==
-            heartbeat_resource_available);
+  // Extract the load information and save it locally.
+  remote_resources.SetLoadResources(
+      ResourceSet(heartbeat_data.resource_load_label,
+                  heartbeat_data.resource_load_capacity));
+  RAY_CHECK(
+      cluster_resource_map_[client_id].GetAvailableResources() == remote_available);
+  RAY_CHECK(
+      cluster_resource_map_[client_id].GetLoadResources() == remote_load);
+
+
+  // Construct cluster resources for the heartbeating client_id & call scheduling policy.
```

Comment:
Sorry to be picky here, but can you make it clear in this comment that the scheduling policy will only consider the client who sent the heartbeat, and that is why you're constructing a temporary `remote_resource_map`?